channeldb/graph.go

```diff
@@ -1712,15 +1718,28 @@ func (c *ChannelGraph) FilterChannelRange(startHeight, endHeight uint32) ([]uint
 
 		// We'll now iterate through the database, and find each
 		// channel ID that resides within the specified range.
-		var cid uint64
 		for k, _ := cursor.Seek(chanIDStart[:]); k != nil &&
 			bytes.Compare(k, chanIDEnd[:]) <= 0; k, _ = cursor.Next() {
 
 			// This channel ID rests within the target range, so
 			// we'll convert it into an integer and add it to our
 			// returned set.
-			cid = byteOrder.Uint64(k)
-			chanIDs = append(chanIDs, cid)
+			rawCid := byteOrder.Uint64(k)
+			cid := lnwire.NewShortChanIDFromInt(rawCid)
+
+			// If we've already added some channel IDs and the
+			// current one has a different height from the last one
+			// added, then increment the index for the next set of
+			// channel IDs.
+			if len(chanIDs) > 0 &&
```

Comment:
this could probably be simplified by using a map from bockheights to channel IDs, then converting this map to a sorted slice before returning.

Reply:
Done.